### PR TITLE
Герасимов - Лабораторная работа #1

### PR DESCRIPTION
**Pull request by GerasimovDG**

### CODE REVIEW

**Comment by GerasimovDG** [comment]
@Trintal

File: code/gerasimov_dmitry_lab1/solving_quadratic_equation.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import math
+
+NO_SOLUTION = "No solution"
+
+
+def solving_quadratic_equation(a, b, c):
```
Comment (okainov) [inline code/gerasimov_dmitry_lab1/solving_quadratic_equation.py:6]: That's really inconsistent. Why does this function return a single number in one case, tuple in another case and a string in the third case?

**Comment by GerasimovDG** [comment]
@l-bat
@mariamosonova

File: code/gerasimov_dmitry_lab1/solving_quadratic_equation.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import math
+
+
+def is_correct_input_type(value):
+    if type(value) is float or type(value) is int:
+        return True
+    return False
```
Comment (l-bat) [inline code/gerasimov_dmitry_lab1/solving_quadratic_equation.py:7]: For me `return  isinstance(value, float) or isinstance(value, int)` is better

File: code/gerasimov_dmitry_lab1/solving_quadratic_equation.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        self.a = a
+        self.b = b
+        self.c = c
+
+    def solution(self):
+        self.answer = list()
+        disc = self.b ** 2 - 4 * self.a * self.c
+        if self.b == 0:
+            if self.c == 0:
+                disc = 0
```
Comment (l-bat) [inline code/gerasimov_dmitry_lab1/solving_quadratic_equation.py:26]: Looks like if `b == 0 and c == 0`, then` disc = self.b ** 2 - 4 * self.a * self.c = 0`

**Comment by l-bat** [comment]
Thank you! 👍

**Comment by GerasimovDG** [comment]
> Thank you! 👍

Thank you for the important comments for me, cause Python is new to me)

**Comment by senyamac** [comment]
👍

**Comment by GerasimovDG** [comment]
@kirill-korniakov

**Comment by mariamosonova** [comment]
👍

**Review by AntonGrishin — APPROVED** [review]

**Pull request merged**